autoload/neomake/utils.vim

```diff
@@ -204,14 +204,23 @@ function! neomake#utils#MakerIsAvailable(ft, maker_name) abort
         " for our purposes
         return 1
     endif
+
     let maker = neomake#GetMaker(a:maker_name, a:ft)
     if empty(maker)
         return 0
     endif
-    if !has_key(s:available_makers, maker.exe)
-        let s:available_makers[maker.exe] = executable(maker.exe)
+
+    if !has_key(s:available_makers, maker.name)
+        if (type(maker.exe) == type(function('tr')))
+            let l:executable = call(maker.exe, [])
+        elseif (type(maker.exe) == type({}))
+            let l:executable = call(maker.exe.fn, [], maker.exe)
+        else
+            let l:executable = maker.exe
+        endif
+        let s:available_makers[maker.name] = executable(l:executable)
     endif
-    return s:available_makers[maker.exe]
+    return s:available_makers[maker.name]
```

Comment:
This somehow reverts https://github.com/jreinert/neomake/commit/2be7919475b1e8a728e32fe8b5e321ee0853aa18?!
Let's maybe just drop the caching here?!